File: MiniDOMFinanziaria/Finanziaria/Vincoli su Tabelle/CTableConstraintCursor.cs
using System;

namespace minidom
{
    public partial class Finanziaria
    {
        public abstract class CTableConstraintCursor : Databases.DBObjectCursor<CTableConstraint>
        {
            private DBCursorStringField m_Espressione = new DBCursorStringField("Espressione");
            private DBCursorField<TypeCode> m_TipoValore = new DBCursorField<TypeCode>("TipoValore");
            private DBCursorField<TableContraints> m_TipoVincolo = new DBCursorField<TableContraints>("TipoVincolo");

            public CTableConstraintCursor()
            {
            }

            protected override Databases.CDBConnection GetConnection()
            {
                return Database;
            }

            protected override Sistema.CModule GetModule()
            {
                return null;
            }

            public abstract override string GetTableName();

            public DBCursorStringField Espressione
            {
                get
                {
                    return m_Espressione;
                }
            }

            public DBCursorField<TypeCode> TipoValore
            {
                get
                {
                    return m_TipoValore;
                }
            }

            public DBCursorField<TableContraints> TipoVincolo
            {
                get
                {
                    return m_TipoVincolo;
                }
            }
        }
    }
}

File: MiniDOMFinanziaria/Finanziaria/Vincoli su Tabelle/CTableConstraintCursor.vb
Imports minidom
Imports minidom.Databases
Imports minidom.Sistema
Imports minidom.Anagrafica



Partial Public Class Finanziaria
 
  

    Public MustInherit Class CTableConstraintCursor
        Inherits DBObjectCursor(Of CTableConstraint)

        Private m_Espressione As New CCursorFieldObj(Of String)("Espressione")
        Private m_TipoValore As New CCursorField(Of System.TypeCode)("TipoValore")
        Private m_TipoVincolo As New CCursorField(Of TableContraints)("TipoVincolo")

        Public Sub New()
        End Sub

        Protected Overrides Function GetConnection() As CDBConnection
            Return Finanziaria.Database
        End Function

        Protected Overrides Function GetModule() As CModule
            Return Nothing
        End Function

        Public MustOverride Overrides Function GetTableName() As String

        Public ReadOnly Property Espressione As CCursorFieldObj(Of String)
            Get
                Return Me.m_Espressione
            End Get
        End Property

        Public ReadOnly Property TipoValore As CCursorField(Of System.TypeCode)
            Get
                Return Me.m_TipoValore
            End Get
        End Property

        Public ReadOnly Property TipoVincolo As CCursorField(Of TableContraints)
            Get
                Return Me.m_TipoVincolo
            End Get
        End Property

    End Class
 
End Class
